src/common/components/signInForm/signInForm.component.js

```diff
@@ -31,6 +33,12 @@ class SignInFormController {
     this.sessionService
       .signIn(this.username, this.password, this.mfa_token, this.trust_device, this.lastPurchaseId)
       .subscribe(() => {
+        const $injector = this.$injector
+        if (!$injector.has('sessionService')) {
+          $injector.loadNewModules(['sessionService'])
```

Comment:
He mentioned here that you don't need to check if it hasn't got the service as it won't add it if it's there. But I guess you knew that already and just wanted to make it more performant.

https://docs.angularjs.org/api/auto/service/$injector#loadNewModules